packages/frontend/src/content/ShowNote.tsx

```diff
@@ -8,40 +8,40 @@ import { HomeButton, Loading, PageContainer } from "../components";
 
 const ShowNote = (props: RouteComponentProps<{ noteId: string }>) => {
   const { noteId } = props;
-  const [isLoading, setIsLoading] = useState(true);
   const [noteContent, setNoteContent] = useState("");
   const [attachment, setAttachment] = useState("");
   const [attachmentURL, setAttachmentURL] = useState("");
+  const [errorMsg, setErrorMsg] = useState("");
+  const [isPending, startTransition] = useTransition();
 
   useEffect(() => {
-    const fetchNote = async (noteId: string) => {
-      setIsLoading(true);
-      const fetchURL = `${GATEWAY_URL}notes/${noteId}`;
+    if (noteId) {
+      startTransition(async () => {
+        const fetchURL = `${GATEWAY_URL}notes/${noteId}`;
 
-      try {
-        const response = await fetch(fetchURL);
-        const data = await response.json();
-        setNoteContent(data.content);
-        if (data.attachment) {
-          setAttachment(data.attachment);
-          setAttachmentURL(await getObjectUrl(data.attachment));
+        try {
+          const response = await fetch(fetchURL);
+          const data = await response.json();
+          setNoteContent(data.content);
+          if (data.attachment) {
+            setAttachment(data.attachment);
+            setAttachmentURL(await getObjectUrl(data.attachment));
+          }
+        } catch (error) {
+          // Navigate to 404 page, as noteId probably not present
+          navigate("/404");
         }
-      } catch (error) {
-        // Navigate to 404 page, as noteId probably not present
-        navigate("/404");
-      } finally {
-        setIsLoading(false);
-      }
-    };
-    fetchNote(noteId || "");
+      });
+    }
   }, [noteId]);
 
   return (
     <PageContainer header={<HomeButton />}>
-      {isLoading ? (
+      {isPending ? (
         <Loading />
       ) : (
         <form>
+          {errorMsg && <div className="alert alert-danger">{errorMsg}</div>}
```

Comment:
The `errorMsg` is not set in this component, do it's not going to be displayed.
This can be removed.
```suggestion
```